bin/post_process_sce.R

```diff
@@ -84,6 +105,37 @@ if(all(is.na(sce$miQC_pass))){
 # add min gene cutoff to metadata
 metadata(sce)$min_gene_cutoff <- opt$gene_cutoff
 
+# read in ADT feature barcode file, if present
+if (!(is.null(opt$adt_barcode_file))) {
+
+  # Create data frame of ADTs and their target types
+  adt_barcode_df <- readr::read_tsv(opt$adt_barcode_file, col_names=FALSE)
+
+  # Check if 3rd column is present and assign column names, values assuming
+  #   that all ADTs are targets if there is no 3rd column
+  if (ncol(adt_barcode_df) == 2) {
+    colnames(adt_barcode_df) <- c("name", "barcode")
+    adt_barcode_df$target_type <- "target"
+  } else if (ncol(adt_barcode_df) == 3) {
+    colnames(adt_barcode_df) <- c("name", "barcode", "target_type")
+  } else {
+    stop("The ADT feature barcode file must have either 2 or 3 columns.")
+  }
```

Comment:
Do we really care if there are more than three columns? If not (and I don't think we should care for future robustness), we can simplify the reading here.

```suggestion
  # Create data frame of ADTs and their target types
  adt_barcode_df <- readr::read_tsv(
    opt$adt_barcode_file, 
    col_names = c("name", "barcode", "target_type")
    # if only 2 columns exist, only the first two col_names will be used
  ) 
    
  if (!"target_type" %in% names(adt_barcode_df)) {
    adt_barcode_df$target_type <- "target"
  } 
```

Reply:
I'll implement this but can't accept suggestion without stashing in-progress changes, and it's easier to copy/paste...

Reply:
Noting this is done in https://github.com/AlexsLemonade/scpca-nf/pull/312/commits/0b38c3f48a6e015a3aeed628253305bdb395a7ac, along with a couple other updates to use unfiltered as well as filtered.